### ISSUE TITLE
ng-model is not binding an image in text-angular in firefox browser

### ISSUE THREAD

**Issue by meta-manish-sharma** [fraywing/textAngular#1201]
I am using text-angular, text is working properly in firefox browser.But when i am inserting image and trying it to bind by ng-model, it's not binding and showing blank in firefox, wheareas working fine in chrome browser.

**Comment by pbassut** [comment]
I don't understand. You want an image to be bound to a variable instead of going to plain html? Note that rendering directives(in your case ngModel) is not possible in textAngular.

**Comment by meta-manish-sharma** [comment]
Hi Patrick Bassut,

Now i am using angular demo(www.textangular.com) screenshots to make you understand. As in angular demo whatever i am doing in textarea it is reflecting back to html output.

**In Chrome browser :-**

1. As text area is empty so html output is also empty. [Click here](https://snag.gy/7ZzUJs.jpg)
2. Now i click on insert image in text area and chose a url of image. [Click here]
(https://snag.gy/HWUmhE.jpg)
3. Image has been inserted in text area and html output is updated with image tag. [Click here](https://snag.gy/PnWc9E.jpg)

So it's working fine with chrome browser.


**In Firefox browser:**

1. Text area is blank as well as html output. [Click here](https://snag.gy/btkBr9.jpg)
2. Now click on insert image and put url of an image. [Click here](https://snag.gy/s36icm.jpg)
3. Now as shown in snapshot, image has been inserted but html output is still blank that means image is not binding in firefox. [Click here](https://snag.gy/gu8win.jpg)

So when i am using ng-model to bind it, it's not working in firefox.
One more thing is also there, all of this happens when we use only  image in textarea, means when we use text with image it works fine in firefox too. [Click here](https://snag.gy/32kiJB.jpg)

**Comment by Aoqidx** [comment]
Hi meta-manish-sharma:
    The same happend to me. If insert a imageUrl in Firefox without other operation, nothing output.  You solve this problem yet?

**Comment by pbassut** [comment]
@meta-manish-sharma Thanks for this. That's a big problem. We'll fix it asap.

**Comment by JoelParke** [comment]
I just checked with with all my recent fixes (my big PR) and it is still an issue, so I am looking into this as well.

**Comment by JoelParke** [comment]
A big discovery:  This only happens IF the html is blank. AND only on Firefox.  Chrome is fine.
Just enter a space and then add image, no problem.
This should be easy to fix -- I think.  Give me about an hour....

**Comment by JoelParke** [comment]
It turns out that Firefox has a bug and then you do a 'insertImage' in our case and there is no content within the 

```
<div id="taTextElementxxxxxxxx" ....>
     <p><br></p>
</div>
```

which is normal for an empty textAngular editor.  

When you do the insertImage on Chrome you get:

```
<div id="taTextElementxxxxxxxx" ....>
     <p><img src="asdf"><br></p>
</div>
```

On Firefox you find:

```
<div id="taTextElementxxxxxxxx" ....>
     <img _moz_dirty="" src="asdf">
</div>
```

Which is very bad.   And it is not clear why Firefox behaves this way.  So the solution that I implemented is to use insertHtml for this case with the expected html of `<img src="asdf">`.

There is some info on the _moz_dirty piece here: [http://stackoverflow.com/questions/2991238/firebug-and-moz-dirty](url)  So it seems like a piece of an old hack in Firefox.

I will commit this fix in a few minutes and put the link here of the commit.

**Comment by JoelParke** [comment]
Fixed with release v1.5.2

**Comment by KryDos** [comment]
Hi, Currently I'm getting pretty similar issue. But in Chrome. 
I was using version 1.5.9 and just updated to 1.5.13 and it didn't help. 

When I try to insert an image, I do not see any changes in the content but my ng-model is updated with empty `<p>` tag. I just tried to test it in Firefox (49.0.2) and it behaves the same as Chrome now (inserts empty <p> tag). 

So, when textAngular content is empty I can't insert an image (it inserts <p></p>). If I will try to do this second time then it works. But if I will click "enter" to move at new line it stop working again and I can insert image only with second try (after <p> tag will be inserted). 

I hope I described it correctly. And thank you that you added issue number in the source code :) I was unable to find this issue via GitHub search.
